cmd/gctsCreateRepository.go

```diff
@@ -0,0 +1,125 @@
+	// error situations should stop execution through log.Entry().Fatal() call which leads to an os.Exit(1) in the end
+	err := createRepository(&config, telemetryData, &c, httpClient)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func createRepository(config *gctsCreateRepositoryOptions, telemetryData *telemetry.CustomData, command execRunner, httpClient piperhttp.Sender) error {
+
+	cookieJar, _ := cookiejar.New(nil)
+	clientOptions := piperhttp.ClientOptions{
+		CookieJar: cookieJar,
+		Username:  config.Username,
+		Password:  config.Password,
+	}
+	httpClient.SetOptions(clientOptions)
+
+	type repoData struct {
+		RID                 string `json:"rid"`
+		Name                string `json:"name"`
+		Role                string `json:"role"`
+		Type                string `json:"type"`
+		VSID                string `json:"vsid"`
+		RemoteRepositoryURL string `json:"url"`
+	}
+
+	type createRequestBody struct {
+		Repository string   `json:"repository"`
+		Data       repoData `json:"data"`
+	}
+
+	reqBody := createRequestBody{
+		Repository: config.Repository,
+		Data: repoData{
+			RID:                 config.Repository,
+			Name:                config.Repository,
+			Role:                config.Role,
+			Type:                config.Type,
+			VSID:                config.VSID,
+			RemoteRepositoryURL: config.RemoteRepositoryURL,
+		},
+	}
+	jsonBody, marshalErr := json.Marshal(reqBody)
+
+	if marshalErr != nil {
+		return fmt.Errorf("creating repository on the ABAP system %v failed: %w", config.Host, marshalErr)
+	}
+
+	header := make(http.Header)
+	header.Set("Content-Type", "application/json")
+	header.Add("Accept", "application/json")
+
+	url := config.Host + "/sap/bc/cts_abapvcs/repository?sap-client=" + config.Client
+
+	resp, httpErr := httpClient.SendRequest("POST", url, bytes.NewBuffer(jsonBody), header, nil)
+
+	defer func() {
+		if resp != nil && resp.Body != nil {
+			resp.Body.Close()
+		}
+	}()
+
+	if resp == nil {
+		return fmt.Errorf("creating repository on the ABAP system %v failed: %w", config.Host, httpErr)
+	}
+
+	bodyText, readErr := ioutil.ReadAll(resp.Body)
+
+	if readErr != nil {
+		return fmt.Errorf("creating repository on the ABAP system %v failed: %w", config.Host, readErr)
+	}
+
+	response, parsingErr := gabs.ParseJSON([]byte(bodyText))
+
+	if parsingErr != nil {
+		return fmt.Errorf("creating repository on the ABAP system %v failed: %w", config.Host, parsingErr)
+	}
+
+	if httpErr != nil {
+		if resp.StatusCode == 500 {
+			if exception, ok := response.Path("exception").Data().(string); ok && exception == "Repository already exists" {
+				log.Entry().
+					WithField("repository", config.Repository).
+					Infof("the repository already exists on the ABAP system %v", config.Host)
+				return nil
+			}
+		}
+		log.Entry().Errorf("a HTTP error occured! Response body: %v", response)
+		return fmt.Errorf("creating repository on the ABAP system %v failed: %w", config.Host, httpErr)
```

Comment:
In other similar cases inside the piper lib we are using exception wrapping, e.g. `errors.Wrap(err, "Some text")`. (`error` is `github.com/pkg/errors`).